### PR TITLE
Allow file-based configuration for Dynatrace v2 exporter

### PR DESCRIPTION
**Pull request by pirgeo**
This PR introduces support for file-based configuration using the `DynatraceFileBasedConfigurationProvider`. It also makes sure changes in the file-based configuration are picked up, without having to restart the app to change configuration. 

This change adds another layer to the auto-configuration that is already available: Instead of falling back to the local OneAgent directly, the the file at `/var/lib/dynatrace/enrichment/endpoint/endpoint.properties` will be checked for configuration. The file will typically be provided by the [Dynatrace Operator](https://github.com/Dynatrace/dynatrace-operator) but can also just be added manually if you'd like to test it locally. I will follow up with a PR updating the documentation in `micrometer-docs` to describe the changes.

### CODE REVIEW

**Comment by pirgeo** [comment]
Hey @jonatan-ivanov and @shakuzen, it would be great if someone could take a look at this PR please, it has been sitting for a while now.
Do you think it's possible to get this into the 1.9.0 release?
Thank you!

**Review by jonatan-ivanov — APPROVED** [review]
Let's merge this and I'm going to add a polish commit.

**Comment by jonatan-ivanov** [comment]
Thanks @pirgeo!

File: implementations/micrometer-registry-dynatrace/src/test/java/com/dynatrace/file/util/FileBasedConfigurationTestHelper.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.dynatrace.file.util;
+
+public class FileBasedConfigurationTestHelper {
+    // This allows overwriting the file location of the file-based configuration for testing.
+    public static void forceOverwriteConfig(String filename) {
+        DynatraceFileBasedConfigurationProvider.getInstance().forceOverwriteConfig(filename);
```
Comment (jonatan-ivanov) [inline implementations/micrometer-registry-dynatrace/src/test/java/com/dynatrace/file/util/FileBasedConfigurationTestHelper.java:22]: I've just realized why is this needed.
Is it possible to make it public or release a test dependency where this is available?

**Comment by jonatan-ivanov** [comment]
@pirgeo I think there are some issues with the Dynatrace file poller, it seems it takes a lot of time on macOS for the file poller to detect the changes.

The tests in this PR were failing for me on my mac (OS X 12.1, x86_64): see [this report](https://ge.micrometer.io/s/g3s2ubcarfaok).
I replaced `Thread::sleep` with awaitility it takes almost 10 seconds on my machine to see the file changes (instead of 10ms that you had). Could you please look into it if anyone else has this issue before?


https://github.com/jonatan-ivanov/micrometer/commit/f4c1c4227bc50d52895ba57d21592c7315b2cf18
